integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataLoadRDD.scala

```diff
@@ -72,8 +76,91 @@ class CarbonNodePartition(rddId: Int, val idx: Int, host: String,
   override def hashCode(): Int = 41 * (41 + rddId) + idx
 }
 
+object CarbonDataLoadRDDFuncs extends Logging{
```

Comment:
Can we have a more meaningful name for this?
I suggest `SparkPartitionLoader` (it loads one spark partition into carbon using kettle), and make it a class instead of object, so that it can keep the storeLocation internally instead of returning it in initialize function. One is using the returned storeLocation variable other than run function.